src/lsqecc/pauli_rotations/rotation.py

```diff
@@ -192,6 +192,43 @@ def from_list(pauli_ops: List[PauliOperator], rotation: Fraction) -> "PauliRotat
             r.change_single_op(i, op)
         return r
 
+    def remove_y_operators(self) -> Tuple[List["PauliRotation"], List["PauliRotation"]]:
+        """Remove Y operators from a Pauli Rotation. Currently supports only pi/8 rotation"""
+        if self.rotation_amount not in (Fraction(1, 8), Fraction(-1, 8)):
+            raise NotImplementedError("Method only supports pi/8 rotations")
+
+        left_rotations: List["PauliRotation"] = list()
+        right_rotations: List["PauliRotation"] = list()
+        y_op_indices = list()
+
+        # Find Y operators and modify them into X operators
+        for index, operator in enumerate(self.ops_list):
+            if operator == PauliOperator.Y:
+                y_op_indices.append(index)
+                self.ops_list[index] = PauliOperator.X
+
+        if not y_op_indices:
+            return left_rotations, right_rotations
```

Comment:
Why are we returning empty lists here? I understand no operator has been removed, but could we not just `return None`?

Reply:
For me it was to keep it consistent with the typing I declared above (and thus keeping mypy happy).